tests/smc/test_smc.py

```diff
@@ -133,6 +133,18 @@ def test_unobserved_categorical(self):
 
         assert np.all(np.median(trace["mu"], axis=0) == [1, 2])
 
+    def test_parallel_custom(self):
+        def _logp(value, mu):
+            return -((value - mu) ** 2)
+
+        def _random(mu, rng=None, size=None):
+            return rng.normal(loc=mu, scale=1, size=size)
+
+        with pm.Model():
+            mu = pm.CustomDist("mu", 0, logp=_logp, random=_random)
```

Comment:
There are two kinds of CustomDist, if instead of `random` you pass `dist` you get a different Op type, that I guess would still fail after this PR

Reply:
Why does it fail with `random` but not `dist`. Testing locally it seems to fail with both for me?

Reply:
I just tested both the code I used in the linked issue and your example below. I still get no errors using `pm.Potential` or `dist` argument. Originally I was using `pymc==5.10.0` but now I tested with `pymc==5.13.0`. Maybe there are differences between Windows and Linux if you're using one?

I'm not completely sure why using `dist` works for me but, based on some quick testing, `DistributionMeta.__new__` is called when e.g. `Normal` is defined and the overloads for builtin distributions are registered there. I'm not well versed in multiprocessing or the way that Python does importing but my hunch is that the worker processes automatically import stuff from `pymc` and the overloads get registered as a side effect. For user-defined `logprob` etc. this is not the case since the registration isn't done during importing.